kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -234,8 +242,117 @@
+      ConfigDef.Type.BOOLEAN;
+  public static final Boolean BIGQUERY_MESSAGE_TIME_PARTITIONING_DEFAULT =                   false;
+  private static final ConfigDef.Importance BIGQUERY_MESSAGE_TIME_PARTITIONING_IMPORTANCE =
+      ConfigDef.Importance.HIGH;
+  private static final String BIGQUERY_MESSAGE_TIME_PARTITIONING_DOC =
+      "Whether or not to use the message time when inserting records. "
+          + "Default uses the connector processing time.";
+
+  public static final String BIGQUERY_PARTITION_DECORATOR_CONFIG =
+      "bigQueryPartitionDecorator";
+  private static final ConfigDef.Type BIGQUERY_PARTITION_DECORATOR_CONFIG_TYPE =
+      ConfigDef.Type.BOOLEAN;
+  //This has been set to true to preserve the existing behavior. However, we can set it to false if field based partitioning is used in BigQuery
+  public static final Boolean BIGQUERY_PARTITION_DECORATOR_DEFAULT =                 true;
+  private static final ConfigDef.Importance BIGQUERY_PARTITION_DECORATOR_IMPORTANCE =
+      ConfigDef.Importance.HIGH;
+  private static final String BIGQUERY_PARTITION_DECORATOR_DOC =
+      "Whether or not to append partition decorator to BigQuery table name when inserting records. "
+          + "Default is true. Setting this to true appends partition decorator to table name (e.g. table$yyyyMMdd depending on the configuration set for bigQueryPartitionDecorator). "
+          + "Setting this to false bypasses the logic to append the partition decorator and uses raw table name for inserts.";
+
+  public static final String BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_CONFIG = "timestampPartitionFieldName";
+  private static final ConfigDef.Type BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_TYPE = ConfigDef.Type.STRING;
+  private static final String BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_DEFAULT = null;
+  private static final ConfigDef.Validator BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_VALIDATOR = new ConfigDef.NonEmptyString();
+  private static final ConfigDef.Importance BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_IMPORTANCE =
+      ConfigDef.Importance.LOW;
+  private static final String BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_DOC =
+      "The name of the field in the value that contains the timestamp to partition by in BigQuery"
+          + " and enable timestamp partitioning for each table. Leave this configuration blank,"
+          + " to enable ingestion time partitioning for each table.";
+
+  public static final String BIGQUERY_CLUSTERING_FIELD_NAMES_CONFIG = "clusteringPartitionFieldNames";
+  private static final ConfigDef.Type BIGQUERY_CLUSTERING_FIELD_NAMES_TYPE = ConfigDef.Type.LIST;
+  private static final List<String> BIGQUERY_CLUSTERING_FIELD_NAMES_DEFAULT = null;
+  private static final ConfigDef.Validator BIGQUERY_CLUSTERING_FIELD_NAMES_VALIDATOR = (name, value) -> {
+    if (value == null) {
+      return;
+    }
+
+    @SuppressWarnings("unchecked")
+    List<String> parsedValue = (List<String>) value;
```

Comment:
Do we have guarantees that this would cast correctly? AFAICT the framework hands a List string object because we set type to list. Wondering if we should be safe and wrap this in a try/catch